.github/workflows/pypi_deploy.yml

```diff
@@ -0,0 +1,72 @@
+# This workflow will upload a Python Package using the PyPi action workflow
+
+name: Upload Python Package
+
+on:
+  push:
+    # only run when a new tag is pushed; and only when pushed to master
+    tags:
+      - '*'    
+    branches: [ master ]
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+       py_version: ["3.6", "3.7", "3.8", "3.6-dev", "3.7-dev", "3.8-dev"] 
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.py_version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.py_version }}
+    - name: Install dependencies
+      run: |
+        pip install --upgrade setuptools pip
+        pip install --upgrade .
+        pip install -r requirements-tests.txt
+        
+    - name: Run pytest with codecoverage
+      run: |
+        coverage run --source=polyply $(which pytest) -vv polyply
+        coverage report --omit='*/bin/pytest'
+        codecov
+  
+  
+  deploy:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@master
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+        
+    - name: Install pypa/build
+      run: >-
+        python -m
+        pip install
+        build
+        --user
```

Comment:
Do you need the --user?

Reply:
this is a straight copy from the python.org see https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/